src/phel/core/sequence-functions.phel

```diff
@@ -303,14 +303,29 @@ collection in map"))
   [(take-while f xs) (drop-while f xs)])
 
 (defn partition
-  "Partition an indexed data structure into tuples of size n. Returns a new array."
+  "Partitions an indexed data structure into arrays of size n. Returns a new array."
   [n xs]
-  (if (< (count xs) n)
-    @[]
+  (if (<= (count xs) n)
+    @[xs]
     (let [res @[]]
       (loop [xs xs]
         (let [[a b] (split-at n xs)]
           (push res a)
           (if (>= (count b) n)
             (recur b)
-            res))))))
+            (do (push res b) res)))))))
+            
+(defn partition
```

Comment:
Isn't this function already defined in line 305? I think this is duplicated, no?

Reply:
OMG. Yes, I forgot to remove it. Nice catch. I am using the Web interface for this PR...